Examples/Algorithms/Geant4GeoModel/src/GdmlDetectorConstruction.cpp

```diff
@@ -0,0 +1,72 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Geant4GeoModel/GdmlDetectorConstruction.hpp"
+
+#include <G4GDMLParser.hh>
+#include <G4LogicalVolume.hh>
+#include <G4PVPlacement.hh>
+#include <G4UnitsTable.hh>
+#include <G4VisAttributes.hh>
+
+#include "GeoMaterial2G4/GeoExtendedMaterial.h"
+#include "GeoModel2G4/ExtParameterisedVolumeBuilder.h"
+#include "GeoModelKernel/GeoBox.h"
+#include "GeoModelKernel/GeoGeometryPluginLoader.h"
+#include "GeoModelKernel/GeoPhysVol.h"
+#include "GeoModelKernel/GeoVGeometryPlugin.h"
+
+using namespace ActsExamples;
+
+GdmlDetectorConstruction::GdmlDetectorConstruction(std::string path)
+    : G4VUserDetectorConstruction(), m_path(std::move(path)) {}
+
+G4VPhysicalVolume* GdmlDetectorConstruction::Construct() {
+  // Load and build the geometry from GeoModel
+  GeoPhysVol* world = CreateTheWorld();
+  GeoGeometryPluginLoader loader;
+  GeoVGeometryPlugin* factory = loader.load(m_path);
+  factory->create(world);
+
+  // build the Geant4 geometry and get an hanlde to the world' volume
+  ExtParameterisedVolumeBuilder* builder =
+      new ExtParameterisedVolumeBuilder("ATLAS");
```

Comment:
Are we assuming GeoModel == ATLAS?

Reply:
I think this is a memory leak. Does it have to be allocated on the heap or could it just be a stack object?